sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableClientTest.java

```diff
@@ -0,0 +1,1057 @@
+        fail();
+    }
+
+    @Test
+    @Tag("SAS")
+    public void generateSasTokenWithMinimumParameters() {
+        final OffsetDateTime expiryTime = OffsetDateTime.of(2021, 12, 12, 0, 0, 0, 0, ZoneOffset.UTC);
+        final TableSasPermission permissions = TableSasPermission.parse("r");
+        final TableSasProtocol protocol = TableSasProtocol.HTTPS_ONLY;
+
+        final TableSasSignatureValues sasSignatureValues =
+            new TableSasSignatureValues(expiryTime, permissions)
+                .setProtocol(protocol)
+                .setVersion(TableServiceVersion.V2019_02_02.getVersion());
+
+        final String sas = tableClient.generateSas(sasSignatureValues);
+
+        assertTrue(
+            sas.startsWith(
+                "sv=2019-02-02"
+                    + "&se=2021-12-12T00%3A00%3A00Z"
+                    + "&tn=" + tableClient.getTableName()
+                    + "&sp=r"
+                    + "&spr=https"
+                    + "&sig="
+            )
+        );
+    }
+
+    @Test
+    @Tag("SAS")
+    public void generateSasTokenWithAllParameters() {
+        final OffsetDateTime expiryTime = OffsetDateTime.of(2021, 12, 12, 0, 0, 0, 0, ZoneOffset.UTC);
+        final TableSasPermission permissions = TableSasPermission.parse("raud");
+        final TableSasProtocol protocol = TableSasProtocol.HTTPS_HTTP;
+
+        final OffsetDateTime startTime = OffsetDateTime.of(2015, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+        final TableSasIpRange ipRange = TableSasIpRange.parse("a-b");
+        final String startPartitionKey = "startPartitionKey";
+        final String startRowKey = "startRowKey";
+        final String endPartitionKey = "endPartitionKey";
+        final String endRowKey = "endRowKey";
+
+        final TableSasSignatureValues sasSignatureValues =
+            new TableSasSignatureValues(expiryTime, permissions)
+                .setProtocol(protocol)
+                .setVersion(TableServiceVersion.V2019_02_02.getVersion())
+                .setStartTime(startTime)
+                .setSasIpRange(ipRange)
+                .setStartPartitionKey(startPartitionKey)
+                .setStartRowKey(startRowKey)
+                .setEndPartitionKey(endPartitionKey)
+                .setEndRowKey(endRowKey);
+
+        final String sas = tableClient.generateSas(sasSignatureValues);
+
+        assertTrue(
+            sas.startsWith(
+                "sv=2019-02-02"
+                    + "&st=2015-01-01T00%3A00%3A00Z"
+                    + "&se=2021-12-12T00%3A00%3A00Z"
+                    + "&tn=" + tableClient.getTableName()
+                    + "&sp=raud"
+                    + "&spk=startPartitionKey"
+                    + "&srk=startRowKey"
+                    + "&epk=endPartitionKey"
+                    + "&erk=endRowKey"
+                    + "&sip=a-b"
+                    + "&spr=https%2Chttp"
+                    + "&sig="
+            )
+        );
+    }
+
+    @Test
+    @Tag("SAS")
```

Comment:
What's the purpose of this tag? Is it used anywhere to selectively run tests?

Reply:
Not at the moment. Other tests had these tags so I thought it was ok to follow the pattern in case we wanted to use them in the future. I can remove them for now and things shouldn't break or anything.